contentcuration/contentcuration/frontend/channelEdit/views/progress/__tests__/progressModal.spec.js

```diff
@@ -1,114 +1,366 @@
+        expect(pageReload).toHaveBeenCalledTimes(1);
+      });
     });
-    wrapper.setMethods({ closeOverlay });
-    wrapper.find('[data-test="refresh"]').trigger('click');
-    expect(closeOverlay).toHaveBeenCalled();
   });
 
-  describe('on cancel task', () => {
-    let wrapper;
+  describe('when syncing', () => {
+    let propsData, storeConfig;
+
     beforeEach(() => {
-      wrapper = makeWrapper({
-        progressPercent() {
-          return 50;
-        },
-      });
+      propsData = {
+        syncing: true,
+      };
+      storeConfig = cloneDeep(STORE_CONFIG);
+      jest
+        .spyOn(storeConfig.modules.currentChannel.getters, 'currentChannel')
+        .mockReturnValue({ publishing: false });
     });
-    it('stop task button should be shown if task is in progress', () => {
-      expect(wrapper.find('[data-test="stop"]').exists()).toBe(true);
+
+    it('progress modal should be displayed', () => {
+      const store = storeFactory(storeConfig);
+      const wrapper = makeWrapper({ propsData, store });
+
+      expect(getProgressModal(wrapper).exists()).toBe(true);
+      expect(getCancelModal(wrapper).exists()).toBe(false);
     });
-    it('clicking stop button should switch window to confirmation window', () => {
-      wrapper.find('[data-test="stop"]').trigger('click');
-      expect(wrapper.vm.step).toBe(2);
+
+    describe('is in progress', () => {
+      let wrapper;
+
+      beforeEach(() => {
+        const store = storeFactory(storeConfig);
+        wrapper = makeWrapper({ propsData, store });
+      });
+
+      it('should display syncing message', () => {
+        expect(getProgressModal(wrapper).text()).toContain(
+          'Channel syncing is in progress, please wait...'
+        );
+      });
+
+      it('should display progress bar', () => {
+        expect(getProgressBar(wrapper).exists()).toBe(true);
+      });
+
+      it("shouldn't display refresh button", () => {
+        expect(getRefreshButton(wrapper).exists()).toBe(false);
+      });
+
+      it('should display stop button', () => {
+        expect(getStopButton(wrapper).exists()).toBe(true);
+      });
+
+      it('clicking stop button should switch to cancel modal', () => {
+        getStopButton(wrapper).trigger('click');
+
+        expect(getCancelModal(wrapper).exists()).toBe(true);
+        expect(getProgressModal(wrapper).exists()).toBe(false);
+      });
+
+      it('clicking cancel button on cancel modal should go back to progress modal', () => {
+        getStopButton(wrapper).trigger('click');
+        expect(getCancelModal(wrapper).exists()).toBe(true);
+
+        getCancelStopButton(wrapper).trigger('click');
+        expect(getCancelModal(wrapper).exists()).toBe(false);
+        expect(getProgressModal(wrapper).exists()).toBe(true);
+      });
+
+      // TODO @MisRob (this logic has been added in another branch)
+      it('clicking confirm stop button on cancel modal should stop syncing and reload the page', () => {});
```

Comment:
I will finish all these TODOs after we merge hotfixes (where @marcellamaki 's  #2964 has been merged) to develop. It would fail now.